05_syntax_lab/04.py

```diff
@@ -1,8 +1,5 @@
-"""
-Write a program that reads lines from the user
-until an empty line is inserted.
-After the user typed in an empty line,
-print all previously inserted lines in reverse
-order (from last to first)
-"""
-
+
+def reverse(line):
+    if line !='': reverse(raw_input())
+    print line    
+reverse(raw_input())

```

Comment:
I love it! though I highly recommend against writing such code in any context other than course exercises, for the reasons @szabgab mentioned :)